ui/src/app/apps/apps.component.ts

```diff
@@ -35,12 +42,21 @@ export class AppsComponent implements OnInit {
     private router: Router ) {
     }
 
+  /**
+   * As soon as the page loads we retrieve a list
+   * of {@link AppRegistration}s.
+   */
   ngOnInit() {
     this.loadAppRegistrations(false);
   }
 
+  /**
+   * Load a paginated list of {@link AppRegistration}s.
+   *
+   * @param reload
+   */
   public loadAppRegistrations(reload: boolean) {
-    this.busy = this.appsService.getApps(reload).subscribe(
+    this.busy.push(this.appsService.getApps(reload).subscribe(
```

Comment:
@jvalkeal It turns out, this was not working correctly anyway. I assumed the busy indicator would work smarter then it does in actuality. Ideally, my page may at random create Observables whose Subscriptions I would like to add to the busy indicator. Therefore, "overlapping Subscriptions" could be a possibility.

Thus, it would have been nice if Busy Indicator could consume a stateful array and upon completion remove Subscriptions from the array. 

Maybe I am off here but just to double-check, I created an issue in the Angular2 Busy issue tracker - https://github.com/devyumao/angular2-busy/issues/77